src/pages/playground/index.tsx

```diff
@@ -23,9 +25,33 @@ import { trpc } from 'lib/client/trpc';
 import type PageWithLayout from 'types/persistentLayout';
 
 const Playground: PageWithLayout = ({}) => {
+  const router = useRouter();
+
+  const isCategories = (
+    value: string | string[] | undefined
+  ): value is trpc['playground']['getAllRequests']['input']['categories'] => {
+    if (value === undefined) {
+      return true;
+    }
+    if (Array.isArray(value)) {
+      value.forEach((category) => {
+        if (!(category in PlaygroundRequestCategory)) return false;
```

Comment:
The forEach callback's return value is ignored, so this likely isn't doing what you want it to do. (The callback, in form of an arrow function, defines an encapsulated execution scope.) There are other useful utility methods that can help here. ([overview of array methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods)) [Array.prototype.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) could help here. Use it like this `return value.some((category) => (category in PlaygroundRequestCategory))`